src/main/java/com/sciamus/contractanalyzer/infrastructure/adapter/mongo/QueryBuilder.java

```diff
@@ -0,0 +1,63 @@
+package com.sciamus.contractanalyzer.infrastructure.adapter.mongo;
+
+import com.sciamus.contractanalyzer.application.ReportFilterParameters;
+import org.springframework.data.mongodb.core.query.Criteria;
+import org.springframework.data.mongodb.core.query.Query;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+
+public class QueryBuilder {
+
+    Query buildQuery(ReportFilterParameters reportFilterParameters) {
+
+        Query query = new Query();
+
+        Criteria andCriteria = new Criteria();
+
+        if (reportFilterParameters.result !=null) {
+            query.addCriteria(Criteria.where("result").is(reportFilterParameters.result));
+        }
+        if (reportFilterParameters.reportBody !=null) {
+            query.addCriteria(Criteria.where("content").regex(reportFilterParameters.reportBody));
+        }
+        if( reportFilterParameters.userName !=null) {
+            query.addCriteria(Criteria.where("userName").is(reportFilterParameters.userName));
+        }
+        if( reportFilterParameters.nameOfCheck !=null) {
+            query.addCriteria(Criteria.where("name").is(reportFilterParameters.nameOfCheck));
+        }
+
+        Criteria from = null;
+        Criteria to = null;
+
+        if(reportFilterParameters.timestampFrom !=null) {
+            if (reportFilterParameters.timestampFrom.length()>10) {
+                LocalDateTime parsed = LocalDateTime.parse(reportFilterParameters.timestampFrom);
+                from = Criteria.where("timestamp").gte(parsed);
+            }
+            else {
+                LocalDate parsed = LocalDate.parse(reportFilterParameters.timestampFrom);
+                from = Criteria.where("timestamp").gte(parsed);
+            }
+        }
+
+        if(reportFilterParameters.timestampTo !=null) {
```

Comment:
## warn

gdy już przekonwertujemy string na datetime'y, nie będzie potrzeby tej ifologii. Po prostu data bez godziny, to data o północy